tests/specs/file.js

```diff
@@ -536,40 +810,38 @@ define(["dist/amd-resolver"], function(Resolver) {
     });
 
 
-    describe("hasProtocol", function() {
-      it("Path with forward slash", function() {
+    describe("when calling hasProtocol", function() {
+      it("a path with a forward slash has a protocol", function() {
         expect(File.hasProtocol("/test")).to.equal(true);
       });
 
-      it("Leading back slash", function() {
+      it("a path with a leading back slash has a protocol", function() {
         expect(File.hasProtocol("\\test")).to.equal(true);
       });
 
-      it("Path with leading dot", function() {
+      it("a path with a leading dot has a protocol", function() {
         expect(File.hasProtocol("./test")).to.equal(true);
       });
 
-      it("Path with 2 leading dot", function() {
+      it("a path with 2 leading dots has a protocol", function() {
         expect(File.hasProtocol("..\\test")).to.equal(true);
       });
 
-      it("Path with no leading slash or dot", function() {
+      it("a path with no leading slash or dot has a protocol", function() {
         expect(File.hasProtocol("test")).to.equal(true);
       });
 
-      it("Path with leading http protocol", function() {
+      it("a path with a leading http protocol does not have a protocol", function() {
```

Comment:
the wording could be tweaked.  But basically if the input sting is just the protocol, then that is not a valid URI.  Primarily because a URI is to define the location of a resource and this `http://` does not tell that...  It tells me what protocol I would use if the URI were valid though. :)